Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java

```diff
@@ -230,12 +230,19 @@ protected void execute() throws InterruptedException, ExecutionException {
                     newHitBox.getMidPoint().getZ());
             test = test.translate(0, -1, 0);
             while (test.toBukkit(world).getBlock().getType().isAir()) {
+                // If we are out of bounds, that is technically air, but can we hover over ground???
+                if (test.getY() < world.getMinHeight()) {
+                    fail(String.format(I18nSupport.getInternationalisedString("Translation - Failed Craft over void")));
+                    return;
+                }
                 test = test.translate(0, -1, 0);
             }
             Material testType = test.toBukkit(world).getBlock().getType();
             if (craft.getType().getMaterialSetProperty(CraftType.FORBIDDEN_HOVER_OVER_BLOCKS).contains(testType)) {
+                // Why is there no return here? Shouldnt there be one?
```

Comment:
No need for this comment

Reply:
It is a serious question i have there.

Reply:
Ok, it seems i confused the lines when i wrote that comment, but anyway. Once the craft goes over a forbidden hover over block, it can't translate at all. Calling fail also pretty much implies that the result is final at that point, but yet the function would continue, so the return is necessary